federation/manifests/federation-controller-manager-deployment.yaml

```diff
@@ -17,16 +17,23 @@ spec:
       - name: ssl-certs
         hostPath:
           path: /etc/ssl/certs
+      - name: kubeconfig
+        secret:
+          secretName: federation-apiserver-kubeconfig
```

Comment:
What happens if this secret does not exist?
Can you verify that it works fine with DeprecatedKubeconfigSecretName.

Its bad if the pod fails to come up if the secret with that name does not exist.
If the pod comes up, just that the value of kubeconfig is empty, then its fine (we will fallback to DeprecatedKubeconfigSecretName) in code.